lnst/Recipes/ENRT/SRIOVNetnsTcRecipe.py

```diff
@@ -204,6 +205,17 @@ def test_wide_configuration(self):
 
         return config
 
+    def perf_test(self, recipe_conf: RecipeConf):
+        result = super().perf_test(recipe_conf)
+        self._dump_tc_rules()
+
+        return result
+
+    def _dump_tc_rules(self):
+        for host in self.matched:
+            for dev in [host.eth0, host.vf_representor_eth0]:
+                host.run(f"tc -s filter show dev {dev.name} ingress")
```

Comment:
Does this include all the rules / config we care about?

I think you could also use `tc qdisc show`, not sure what exactly are you looking for.

Reply:
I'm looking for `Sent hardware N bytes M pkt` lines in following output:
```
2023-10-26 03:54:57           (host1)        -   DEBUG: Executing: "tc -s filter show dev p5p1 ingress"
2023-10-26 03:54:57           (host1)        -   DEBUG: 
    Stdout:
    ----------------------------
    filter protocol arp pref 49150 flower chain 0 
    filter protocol arp pref 49150 flower chain 0 handle 0x1 
      dst_mac ff:ff:ff:ff:ff:ff
      src_mac a6:c1:18:e6:f6:80
      eth_type arp
      skip_sw
      in_hw in_hw_count 1
    	action order 1: mirred (Egress Redirect to device eth0) stolen
    	index 3 ref 1 bind 1 installed 88 sec used 88 sec
    	Action statistics:
    	Sent 0 bytes 0 pkt (dropped 0, overlimits 0 requeues 0) 
    	backlog 0b 0p requeues 0
    	used_hw_stats delayed
    
    filter protocol arp pref 49151 flower chain 0 
    filter protocol arp pref 49151 flower chain 0 handle 0x1 
      dst_mac 82:43:29:03:38:46
      src_mac a6:c1:18:e6:f6:80
      eth_type arp
      skip_sw
      in_hw in_hw_count 1
    	action order 1: mirred (Egress Redirect to device eth0) stolen
    	index 2 ref 1 bind 1 installed 88 sec used 31 sec
    	Action statistics:
    	Sent 180 bytes 3 pkt (dropped 0, overlimits 0 requeues 0) 
    	Sent software 0 bytes 0 pkt
    	Sent hardware 180 bytes 3 pkt
    	backlog 0b 0p requeues 0
    	used_hw_stats delayed
    
    filter protocol ip pref 49152 flower chain 0 
    filter protocol ip pref 49152 flower chain 0 handle 0x1 
      dst_mac 82:43:29:03:38:46
      src_mac a6:c1:18:e6:f6:80
      eth_type ipv4
      skip_sw
      in_hw in_hw_count 1
    	action order 1: mirred (Egress Redirect to device eth0) stolen
    	index 1 ref 1 bind 1 installed 88 sec used 2 sec
    	Action statistics:
    	Sent 195858280 bytes 2967498 pkt (dropped 0, overlimits 0 requeues 0) 
    	Sent software 0 bytes 0 pkt
    	Sent hardware 195858280 bytes 2967498 pkt
    	backlog 0b 0p requeues 0
    	used_hw_stats delayed
    
    ----------------------------
2023-10-26 03:54:57       (localhost)        -    INFO: Result: PASS, What: Job finished: type(shell), host(host1), netns(None), 'tc -s filter show dev p5p1 ingress'
2023-10-26 03:54:57       (localhost)        -   DEBUG: Host host1 executing job 32: type(shell), host(host1), netns(None), 'tc -s filter show dev eth0 ingress'
2023-10-26 03:54:57       (localhost)        -   DEBUG: Job.what = 'tc -s filter show dev eth0 ingress'
2023-10-26 03:54:57       (localhost)        -    INFO: Result: PASS, What: Job started: type(shell), host(host1), netns(None), 'tc -s filter show dev eth0 ingress'
2023-10-26 03:54:57           (host1)        -   DEBUG: Running job 32 with pid "15388"
2023-10-26 03:54:57           (host1)        -   DEBUG: Executing: "tc -s filter show dev eth0 ingress"
2023-10-26 03:54:57           (host1)        -   DEBUG: 
    Stdout:
    ----------------------------
    filter protocol arp pref 49150 flower chain 0 
    filter protocol arp pref 49150 flower chain 0 handle 0x1 
      dst_mac ff:ff:ff:ff:ff:ff
      src_mac 82:43:29:03:38:46
      eth_type arp
      skip_sw
      in_hw in_hw_count 1
    	action order 1: mirred (Egress Redirect to device p5p1) stolen
    	index 6 ref 1 bind 1 installed 88 sec used 85 sec
    	Action statistics:
    	Sent 60 bytes 1 pkt (dropped 0, overlimits 0 requeues 0) 
    	Sent software 0 bytes 0 pkt
    	Sent hardware 60 bytes 1 pkt
    	backlog 0b 0p requeues 0
    	used_hw_stats delayed
    
    filter protocol arp pref 49151 flower chain 0 
    filter protocol arp pref 49151 flower chain 0 handle 0x1 
      dst_mac a6:c1:18:e6:f6:80
      src_mac 82:43:29:03:38:46
      eth_type arp
      skip_sw
      in_hw in_hw_count 1
    	action order 1: mirred (Egress Redirect to device p5p1) stolen
    	index 5 ref 1 bind 1 installed 88 sec used 31 sec
    	Action statistics:
    	Sent 120 bytes 2 pkt (dropped 0, overlimits 0 requeues 0) 
    	Sent software 0 bytes 0 pkt
    	Sent hardware 120 bytes 2 pkt
    	backlog 0b 0p requeues 0
    	used_hw_stats delayed
    
    filter protocol ip pref 49152 flower chain 0 
    filter protocol ip pref 49152 flower chain 0 handle 0x1 
      dst_mac a6:c1:18:e6:f6:80
      src_mac 82:43:29:03:38:46
      eth_type ipv4
      skip_sw
      in_hw in_hw_count 1
    	action order 1: mirred (Egress Redirect to device p5p1) stolen
    	index 4 ref 1 bind 1 installed 88 sec used 2 sec
    	Action statistics:
    	Sent 184486380989 bytes 121853980 pkt (dropped 0, overlimits 0 requeues 0) 
    	Sent software 0 bytes 0 pkt
    	Sent hardware 184486380989 bytes 121853980 pkt
    	backlog 0b 0p requeues 0
    	used_hw_stats delayed
    
    ----------------------------
2023-10-26 03:54:58       (localhost)        -    INFO: Result: PASS, What: Job finished: type(shell), host(host1), netns(None), 'tc -s filter show dev eth0 ingress'
2023-10-26 03:54:58       (localhost)        -   DEBUG: Host host2 executing job 30: type(shell), host(host2), netns(None), 'tc -s filter show dev p5p1 ingress'
2023-10-26 03:54:58       (localhost)        -   DEBUG: Job.what = 'tc -s filter show dev p5p1 ingress'
2023-10-26 03:54:58       (localhost)        -    INFO: Result: PASS, What: Job started: type(shell), host(host2), netns(None), 'tc -s filter show dev p5p1 ingress'
2023-10-26 03:54:58           (host2)        -   DEBUG: Running job 30 with pid "15300"
2023-10-26 03:54:58           (host2)        -   DEBUG: Executing: "tc -s filter show dev p5p1 ingress"
2023-10-26 03:54:58           (host2)        -   DEBUG: 
    Stdout:
    ----------------------------
    filter protocol arp pref 49150 flower chain 0 
    filter protocol arp pref 49150 flower chain 0 handle 0x1 
      dst_mac ff:ff:ff:ff:ff:ff
      src_mac 82:43:29:03:38:46
      eth_type arp
      skip_sw
      in_hw in_hw_count 1
    	action order 1: mirred (Egress Redirect to device eth0) stolen
    	index 3 ref 1 bind 1 installed 88 sec used 85 sec
    	Action statistics:
    	Sent 60 bytes 1 pkt (dropped 0, overlimits 0 requeues 0) 
    	Sent software 0 bytes 0 pkt
    	Sent hardware 60 bytes 1 pkt
    	backlog 0b 0p requeues 0
    	used_hw_stats delayed
    
    filter protocol arp pref 49151 flower chain 0 
    filter protocol arp pref 49151 flower chain 0 handle 0x1 
      dst_mac a6:c1:18:e6:f6:80
      src_mac 82:43:29:03:38:46
      eth_type arp
      skip_sw
      in_hw in_hw_count 1
    	action order 1: mirred (Egress Redirect to device eth0) stolen
    	index 2 ref 1 bind 1 installed 88 sec used 31 sec
    	Action statistics:
    	Sent 120 bytes 2 pkt (dropped 0, overlimits 0 requeues 0) 
    	Sent software 0 bytes 0 pkt
    	Sent hardware 120 bytes 2 pkt
    	backlog 0b 0p requeues 0
    	used_hw_stats delayed
    
    filter protocol ip pref 49152 flower chain 0 
    filter protocol ip pref 49152 flower chain 0 handle 0x1 
      dst_mac a6:c1:18:e6:f6:80
      src_mac 82:43:29:03:38:46
      eth_type ipv4
      skip_sw
      in_hw in_hw_count 1
    	action order 1: mirred (Egress Redirect to device eth0) stolen
    	index 1 ref 1 bind 1 installed 88 sec used 2 sec
    	Action statistics:
    	Sent 184486380989 bytes 121853980 pkt (dropped 0, overlimits 0 requeues 0) 
    	Sent software 0 bytes 0 pkt
    	Sent hardware 184486380989 bytes 121853980 pkt
    	backlog 0b 0p requeues 0
    	used_hw_stats delayed
    
    ----------------------------
2023-10-26 03:54:58       (localhost)        -   DEBUG: Waiting for Job 30 on Host host2 for 60 seconds.
2023-10-26 03:54:58       (localhost)        -    INFO: Result: PASS, What: Job finished: type(shell), host(host2), netns(None), 'tc -s filter show dev p5p1 ingress'
2023-10-26 03:54:58       (localhost)        -   DEBUG: Condition passed, disabling timeout alarm
2023-10-26 03:54:58       (localhost)        -   DEBUG: Host host2 executing job 31: type(shell), host(host2), netns(None), 'tc -s filter show dev eth0 ingress'
2023-10-26 03:54:58       (localhost)        -   DEBUG: Job.what = 'tc -s filter show dev eth0 ingress'
2023-10-26 03:54:58       (localhost)        -    INFO: Result: PASS, What: Job started: type(shell), host(host2), netns(None), 'tc -s filter show dev eth0 ingress'
2023-10-26 03:54:58           (host2)        -   DEBUG: Running job 31 with pid "15302"
2023-10-26 03:54:58           (host2)        -   DEBUG: Executing: "tc -s filter show dev eth0 ingress"
2023-10-26 03:54:58           (host2)        -   DEBUG: 
    Stdout:
    ----------------------------
    filter protocol arp pref 49150 flower chain 0 
    filter protocol arp pref 49150 flower chain 0 handle 0x1 
      dst_mac ff:ff:ff:ff:ff:ff
      src_mac a6:c1:18:e6:f6:80
      eth_type arp
      skip_sw
      in_hw in_hw_count 1
    	action order 1: mirred (Egress Redirect to device p5p1) stolen
    	index 6 ref 1 bind 1 installed 88 sec used 88 sec
    	Action statistics:
    	Sent 0 bytes 0 pkt (dropped 0, overlimits 0 requeues 0) 
    	backlog 0b 0p requeues 0
    	used_hw_stats delayed
    
    filter protocol arp pref 49151 flower chain 0 
    filter protocol arp pref 49151 flower chain 0 handle 0x1 
      dst_mac 82:43:29:03:38:46
      src_mac a6:c1:18:e6:f6:80
      eth_type arp
      skip_sw
      in_hw in_hw_count 1
    	action order 1: mirred (Egress Redirect to device p5p1) stolen
    	index 5 ref 1 bind 1 installed 88 sec used 32 sec
    	Action statistics:
    	Sent 180 bytes 3 pkt (dropped 0, overlimits 0 requeues 0) 
    	Sent software 0 bytes 0 pkt
    	Sent hardware 180 bytes 3 pkt
    	backlog 0b 0p requeues 0
    	used_hw_stats delayed
    
    filter protocol ip pref 49152 flower chain 0 
    filter protocol ip pref 49152 flower chain 0 handle 0x1 
      dst_mac 82:43:29:03:38:46
      src_mac a6:c1:18:e6:f6:80
      eth_type ipv4
      skip_sw
      in_hw in_hw_count 1
    	action order 1: mirred (Egress Redirect to device p5p1) stolen
    	index 4 ref 1 bind 1 installed 88 sec used 2 sec
    	Action statistics:
    	Sent 195858280 bytes 2967498 pkt (dropped 0, overlimits 0 requeues 0) 
    	Sent software 0 bytes 0 pkt
    	Sent hardware 195858280 bytes 2967498 pkt
    	backlog 0b 0p requeues 0
    	used_hw_stats delayed
    
    ----------------------------
```

All rules that we've added are printed there.

Reply:
@Axonis can we resolve this?